packages/workers-shared/asset-server-worker/src/index.ts

```diff
@@ -21,14 +20,13 @@ export default {
 		const url = new URL(request.url);
 		const { pathname } = url;
 
-		const isLocalDevContext = new Uint8Array(ASSETS_MANIFEST).at(0) === 1;
-		const assetsManifest = isLocalDevContext
-			? new NoopAssetsManifest()
-			: new AssetsManifest(ASSETS_MANIFEST);
-		const assetEntry = await assetsManifest.get(pathname);
-
-		const content = await ASSETS_KV_NAMESPACE.get(assetEntry);
+		const assetsManifest = new AssetsManifest(ASSETS_MANIFEST);
+		const assetKey = await assetsManifest.get(pathname);
+		if (!assetKey) {
+			return new Response("Not Found", { status: 404 });
+		}
 
+		const content = await ASSETS_KV_NAMESPACE.get(assetKey);
 		if (!content) {
 			return new Response("Not Found", { status: 404 });
```

Comment:
Could even go so far as to 500 error here, since we should never have an assetKey from the manifest that doesn't already exist in KV, right? @GregBrimble

Reply:
I am handling that [here](https://github.com/cloudflare/workers-sdk/pull/6539/files#diff-67a90fb405dc23a85bdeb2be30130464cbfcc7c742e4f48cab05bee45f8d36dbR42)